contracts/Strategy.sol

```diff
@@ -92,46 +115,98 @@ contract Strategy is BaseStrategy {
             _rewards,
             _keeper,
             _bentoBox,
-            _kashiPair
+            _kashiPairs,
+            _pids
         );
 
         emit Cloned(newStrategy);
     }
 
-    function _initializeStrat(address _bentoBox, address _kashiPair) internal {
+    function _initializeStrat(
+        address _bentoBox,
+        address[] memory _kashiPairs,
+        uint256[] memory _pids
+    ) internal {
         require(
-            address(kashiPair) == address(0),
+            address(bentoBox) == address(0),
             "strategy already initialized"
         );
-        require(address(IKashiPair(_kashiPair).bentoBox()) == _bentoBox);
-        require(address(IKashiPair(_kashiPair).asset()) == address(want));
+        require(_kashiPairs.length <= maxPairs, "exceeded maxPairs");
+        require(
+            _kashiPairs.length == _pids.length,
+            "kashiPairs and pids must be same length"
+        );
 
         bentoBox = IBentoBox(_bentoBox);
-        kashiPair = IKashiPair(_kashiPair);
+
+        for (uint256 i = 0; i < _kashiPairs.length; i++) {
+            kashiPairs.push(
+                KashiPairInfo(IKashiPair(_kashiPairs[i]), _pids[i])
+            );
+            require(
+                address(kashiPairs[i].kashiPair.bentoBox()) == _bentoBox,
+                "bento does not match"
+            );
+            require(
+                address(kashiPairs[i].kashiPair.asset()) == address(want),
+                "asset does not match want"
+            );
+
+            if (_pids[i] != 0) {
+                require(
+                    address(masterChef.poolInfo(_pids[i]).lpToken) ==
+                        _kashiPairs[i],
+                    "incorrect pid"
+                );
+
+                IERC20(_kashiPairs[i]).safeApprove(
+                    address(masterChef),
+                    type(uint256).max
+                );
+            }
+        }
 
         want.safeApprove(_bentoBox, type(uint256).max);
+        sushi.safeApprove(address(sushiRouter), type(uint256).max);
+
+        // Initialize the swap path
+        path = new address[](2);
+        path[0] = address(sushi);
+        path[1] = address(want);
     }
 
     function name() external view override returns (string memory) {
-        return
-            string(
-                abi.encodePacked(
-                    "StrategyKashiLender(",
-                    kashiPair.symbol(),
-                    ")"
-                )
-            );
+        return string(abi.encodePacked("StrategyKashiMultiPairLender"));
```

Comment:
if you are not going to concatenate, just do `return "StrategyKashiMultiPairLender"`

Reply:
I wonder if we should add a `strategyName` variable to track pairs 🤔

Reply:
Done ✅